SPEC/STATS.md

```diff
@@ -25,12 +25,16 @@ Where:
   - `poll` is used to print bandwidth at an interval.
   - `interval` is the time interval to wait between updating output, if `poll` is true.
 
-`callback` must follow `function (err, stats) {}` signature, where `err` is an error if the operation was not successful. `stats` is an Object containing the following keys:
+If `poll` is `true`, then `callback` must follow `function (err, stream) {}` signature, where `err` is an error if the operation was not successful and `stream` is a Pull Stream where you can listen to the event `data` with a listener that must follow `function (stat) {}` signature.
```

Comment:
@hacdias this wasn't necessarily what I was hoping for here. Follow the example of https://github.com/ipfs/interface-ipfs-core/blob/master/SPEC/FILES.md#files-api and create two api calls.

- `ipfs.stats.bw` - single requests
- `ipfs.stats.bwPoll` - return a stream

Reply:
`bwPoll` returns which _kind_ of stream?

Reply:
- `ipfs.stats.bwPollPullStream`
- `ipfs.stats.bwPollReadableStream`

🙂

Reply:
I'd just suggest to take the 'Poll' out of the equation and use this three functions:

- `stats.bw`
- `stats.bwPullStream`
- `stats.bwReadableStream`

Reply:
What do you think @diasdavid?

Reply:
![image](https://user-images.githubusercontent.com/5447088/35860124-ba66e704-0b3a-11e8-9a16-7e7eabd87ec6.png)

I suggest to structure the SPEC like that instead of creating different entries for each function and add examples for each of them. In FILES spec, we have:

![image](https://user-images.githubusercontent.com/5447088/35860143-da3be282-0b3a-11e8-80c3-ba92cb2d8e62.png)

Which is not accurate: we have one API endpoint which is `cat`. The other functions should be called variants or something like that and not separated in the SPEC. Why? Duplicated information (on the parameters for example) and then we have that `GO - WIP` thing which may make users think there will be `catPullStream` or `[insert your command here]ReadableStream` on Go. That's something I'd like to work on after finishing this Pull Request and make this interface documentation a little better. What do you think @diasdavid?

Reply:
@hacdias let's avoid painting the 🚲🏡 in this PR. tl;dr; they are top level functions, they should be listed in the table of contents and have urls to it.